### ISSUE TITLE
retry logic overrides uri to '/' caused undefined behaviour

### ISSUE THREAD

**Issue by oashi-masako** [sofastack/sofa-mosn#780]
### Describe the bug

```json
{
  "servers": [
    {
      "default_log_path": "stdout",
      "default_log_level": "TRACE",
      "listeners": [
        {
          "name": "httpbin",
          "address": "0.0.0.0:8888",
          "bind_port": true,
          "log_path": "stdout",
          "filter_chains": [
            {
              "filters": [
                {
                  "type": "proxy",
                  "config": {
                    "downstream_protocol": "Http1",
                    "upstream_protocol": "Http1",
                    "router_config_name": "httpbin"
                  }
                },
                {
                  "type": "connection_manager",
                  "config": {
                    "router_config_name": "httpbin",
                    "virtual_hosts": [
                      {
                        "name": "httpbin",
                        "domains": [
                          "*"
                        ],
                        "routers": [
                          {
                            "match": {
                              "prefix": "/"
                            },
                            "route": {
                              "cluster_name": "httpbin",
                              "retry_policy": {
                                "retry_on": true,
                                "retry_timeout": "0s"
                              },
                              "timeout": "0s"
                            }
                          }
                        ]
                      }
                    ]
                  }
                }
              ]
            }
          ]
        }
      ]
    }
  ],
  "cluster_manager": {
    "clusters": [
      {
        "Name": "httpbin",
        "type": "SIMPLE",
        "lb_type": "LB_ROUNDROBIN",
        "max_request_per_conn": 1024,
        "conn_buffer_limit_bytes": 32768,
        "hosts": [
          {
            "address": "172.16.243.128:65535"
          }
        ]
      }
    ]
  }
}
```

The `172.16.243.128:65535` is httpbin service instance. MOSN expose it with `0.0.0.0:8888`.

```shell
curl -v http://127.0.0.1:8888/status/500
```

### Expected behavior

Get 500 responsed.

### Actual behavior

Got 200 responsed.

### Steps to reproduce

Use the config above.

### Minimal yet complete reproducer code (or GitHub URL to code)

https://github.com/sofastack/sofa-mosn/blob/93fbacb4b439ec98999812e72d45e7614b70ab85/pkg/stream/http/stream.go#L731-L742

I notice that in the process above, the uri had been deleted, when doing retry, mosn request '/' rather than '/status/500'.

### Environment
+ SOFAMosn Version

P.S. I'm very curious about that, is mosn really running in production for years or at least months what serve tens of thousands requests?

**Comment by taoyuanyuan** [comment]
wow, it‘s a bug, Thank you for your feedback!
Mosn has been running for half a year in production, and our scene requires no retry, so we didn't find this problem. 
bugfix coming soon.

**Comment by nejisama** [comment]
fixed in #786
